reconcile/utils/mr/auto_promoter.py

```diff
@@ -32,10 +36,20 @@ class ParentSaasConfigPromotion:
 class AutoPromoter(MergeRequestBase):
     name = "auto_promoter"
 
-    def __init__(self, promotions):
-        self.promotions = [p for p in promotions if p]
+    def __init__(
+        self, promotions: Union[Sequence[Promotion], Sequence[dict[str, Any]]]
+    ):
+        # !!! Attention !!!
+        # AutoPromoter is also initialized with promitions as dict by 'gitlab_mr_sqs_consumer'
+        # loaded from SQS message body, therefore self.promotions must be json serializable
+        self.promotions = [
+            p.dict() if isinstance(p, Promotion) else p for p in promotions
```

Comment:
Not sure, but would it be safer here to do `dict(by_alias=True)`?

Reply:
`Promotion` isn't a qenerate generated GQL class, and it doesn't use aliases.

Reply:
Was wondering because it has some Protocol class as one attribute. Can't that potentially be a problem?

Reply:
Good point! Thanks for that! 🙇 The `json.dump` was working, but not the `json.load` :(
I reworked this stuff and introduced a `SaasPromotion` Protocol, set `by_alias`, and extended the test fixtures.